shell/server/pre-meteor.js

```diff
@@ -210,6 +210,25 @@ function checkMagic(buf, magic) {
   return true;
 }
 
+function serveSelfTest(req, res) {
+  inMeteor(() => {
+    if (req.method === 'GET' &&
+        req.url === '/') {
+      const content = new Buffer("Self-test OK.");
+      res.writeHead(200, {
+        'Content-Type': 'text/plain',
+        'Content-Length': content.length,
+        'Access-Control-Allow-Origin': process.env.ROOT_URL,
```

Comment:
You could make this `*`, which would avoid the need to change it later when we support multiple `ROOT_URL`s. I don't see any danger in allowing all origins.